x-pack/plugins/ml/public/application/data_frame_analytics/pages/components/analytics_selector/analytics_id_selector.tsx

```diff
@@ -0,0 +1,315 @@
+    }),
+    sortable: false,
+    truncateText: false,
+    'data-test-subj': 'mlAnalyticsSelectorColumnDescription',
+  },
+];
+
+interface Props {
+  setAnalyticsId: React.Dispatch<React.SetStateAction<AnalyticsSelectorIds | undefined>>;
+  jobsOnly?: boolean;
+}
+
+export function AnalyticsIdSelector({ setAnalyticsId, jobsOnly = false }: Props) {
+  const [selected, setSelected] = useState<
+    { model_id?: string; job_id?: string; analysis_type?: string } | undefined
+  >();
+  const [analyticsJobs, setAnalyticsJobs] = useState<DataFrameAnalyticsConfig[]>([]);
+  const [trainedModels, setTrainedModels] = useState<TrainedModelConfigResponse[]>([]);
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState<boolean>(true);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const { displayErrorToast } = useToastNotificationService();
+  const trainedModelsApiService = useTrainedModelsApiService();
+  const {
+    dataFrameAnalytics: { getDataFrameAnalytics },
+  } = useMlApiContext();
+
+  function renderTabs() {
+    return <EuiTabbedContent size="s" tabs={tabs} initialSelectedTab={tabs[0]} />;
+  }
+
+  async function fetchAnalyticsJobs() {
+    setIsLoading(true);
+    try {
+      const { data_frame_analytics: dataFrameAnalytics }: GetDataFrameAnalyticsResponse =
+        await getDataFrameAnalytics();
+      setAnalyticsJobs(dataFrameAnalytics);
+    } catch (e) {
+      console.error('Error fetching analytics', e); // eslint-disable-line
+      displayErrorToast(
+        e,
+        i18n.translate('xpack.ml.analyticsSelector.analyticsFetchErrorMessage', {
+          defaultMessage: 'An error occurred fetching analytics jobs. Refresh and try again.',
+        })
+      );
+    }
+    setIsLoading(false);
+  }
+
+  async function fetchAnalyticsModels() {
+    setIsLoading(true);
+    try {
+      const response = await trainedModelsApiService.getTrainedModels(undefined, {
+        size: 1000,
+      });
+      setTrainedModels(response);
+    } catch (e) {
+      console.error('Error fetching trained models', e); // eslint-disable-line
+      displayErrorToast(
+        e,
+        i18n.translate('xpack.ml.analyticsSelector.trainedModelsFetchErrorMessage', {
+          defaultMessage: 'An error occurred fetching trained models. Refresh and try again.',
+        })
+      );
+    }
+    setIsLoading(false);
+  }
+
+  function closeFlyout() {
+    setIsFlyoutVisible(false);
+  }
+
+  // Fetch analytics jobs and models on flyout open
+  useEffect(() => {
+    fetchAnalyticsJobs();
+    if (jobsOnly === false) {
+      fetchAnalyticsModels();
+    }
+  }, []);
+
+  const applySelection = useCallback(() => {
+    if (selected !== undefined) {
+      setAnalyticsId(selected);
+    }
+    closeFlyout();
+  }, [selected?.model_id, selected?.job_id]);
+
+  const pagination = {
+    initialPageSize: 5,
+    pageSizeOptions: [3, 5, 8],
+  };
+
+  const selectionValue = {
+    selectable: (item: TableItem) => {
```

Comment:
We should prevent selection for models which aren't linked to DFA jobs - this would consistent with how the actions in the Trained Models page work.